ui/src/authors/containers/CitationSummaryContainer.jsx

```diff
@@ -0,0 +1,58 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import { Map } from 'immutable';
+
+import { fetchCitationSummary } from '../../actions/citations';
+import CitationSummaryTable from '../components/CitationSummaryTable/CitationSummaryTable';
+import LoadingOrChildren from '../../common/components/LoadingOrChildren';
+import ErrorAlertOrChildren from '../../common/components/ErrorAlertOrChildren';
+import { ErrorPropType } from '../../common/propTypes';
+
+class CitationSummaryContainer extends Component {
+  componentDidMount() {
+    this.fetchCitationSummary();
+  }
+
+  componentDidUpdate(prevProps) {
+    const prevSearchQuery = prevProps.searchQuery;
+    const { searchQuery } = this.props;
+    if (searchQuery !== prevSearchQuery) {
+      this.fetchCitationSummary();
+    }
+  }
+
+  fetchCitationSummary() {
+    const { searchQuery, dispatch } = this.props;
+    dispatch(fetchCitationSummary(searchQuery));
+  }
+
+  render() {
+    const { citationSummary, loadingCitationSummary, error } = this.props;
+    return (
+      <LoadingOrChildren loading={loadingCitationSummary}>
+        <ErrorAlertOrChildren error={error}>
+          <CitationSummaryTable citationSummary={citationSummary} />
```

Comment:
could you make default `citationSummary`, `null`. Then do `citationSummary  && ...` here.

Reply:
hm, but I made citationSummary required, should it not be required?